2025-02-19 14:26:25,842 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-19 14:28:17,748 ERROR: Exception on /edit_profile [POST] [in C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\flask\app.py:875]
Traceback (most recent call last):
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1964, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\default.py", line 942, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.IntegrityError: UNIQUE constraint failed: user.username

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\app\routes.py", line 87, in edit_profile
    db.session.commit()
    ~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\scoping.py", line 599, in commit
    return self._proxied.commit()
           ~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\session.py", line 2032, in commit
    trans.commit(_to_root=True)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "<string>", line 2, in commit
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\session.py", line 1313, in commit
    self._prepare_impl()
    ~~~~~~~~~~~~~~~~~~^^
  File "<string>", line 2, in _prepare_impl
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\session.py", line 1288, in _prepare_impl
    self.session.flush()
    ~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\session.py", line 4353, in flush
    self._flush(objects)
    ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\session.py", line 4488, in _flush
    with util.safe_reraise():
         ~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\util\langhelpers.py", line 146, in __exit__
    raise exc_value.with_traceback(exc_tb)
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\session.py", line 4449, in _flush
    flush_context.execute()
    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\unitofwork.py", line 466, in execute
    rec.execute(self)
    ~~~~~~~~~~~^^^^^^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\unitofwork.py", line 642, in execute
    util.preloaded.orm_persistence.save_obj(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self.mapper,
        ^^^^^^^^^^^^
        uow.states_for_mapper_hierarchy(self.mapper, False, False),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        uow,
        ^^^^
    )
    ^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\persistence.py", line 85, in save_obj
    _emit_update_statements(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        base_mapper,
        ^^^^^^^^^^^^
    ...<3 lines>...
        update,
        ^^^^^^^
    )
    ^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\orm\persistence.py", line 912, in _emit_update_statements
    c = connection.execute(
        statement, multiparams, execution_options=execution_options
    )
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1416, in execute
    return meth(
        self,
        distilled_parameters,
        execution_options or NO_OPTIONS,
    )
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\sql\elements.py", line 516, in _execute_on_connection
    return connection._execute_clauseelement(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self, distilled_params, execution_options
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1638, in _execute_clauseelement
    ret = self._execute_context(
        dialect,
    ...<8 lines>...
        cache_hit=cache_hit,
    )
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1843, in _execute_context
    return self._exec_single_context(
           ~~~~~~~~~~~~~~~~~~~~~~~~~^
        dialect, context, statement, parameters
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1983, in _exec_single_context
    self._handle_dbapi_exception(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        e, str_statement, effective_parameters, cursor, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\base.py", line 2352, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\base.py", line 1964, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Olga\PycharmProjects\Flask_book\.venv\Lib\site-packages\sqlalchemy\engine\default.py", line 942, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: user.username
[SQL: UPDATE user SET username=?, about_me=? WHERE user.id = ?]
[parameters: ('olga', 'fvdfvdvdfv', 1)]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
2025-02-21 13:31:29,329 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-21 13:31:51,567 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 12:28:23,546 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 12:39:52,459 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 16:21:19,719 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:15:03,814 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:19:22,586 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:20:44,426 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:25:01,993 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:25:29,308 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:27:54,509 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:29:18,652 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:29:49,522 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:30:22,820 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 17:30:53,168 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
2025-02-24 18:38:56,246 INFO: My project [in C:\Users\Olga\PycharmProjects\Flask_book\app\__init__.py:41]
